packages/compiler/lib/decorators.ts

```diff
@@ -813,3 +813,62 @@ export function isDeprecated(program: Program, type: Type): boolean {
 export function getDeprecated(program: Program, type: Type): string | undefined {
   return program.stateMap(deprecatedKey).get(type);
 }
+
+const overloadedByKey = Symbol("overloadedByKey");
+const overloadsOperationKey = Symbol("overloadsOperation");
+
+/**
+ * `@overload` - Indicate that the target overloads (specializes) the overloads type.
+ * @param context DecoratorContext
+ * @param target The specializing operation declaration
+ * @param overloads The operation to be overloaded.
+ */
+export function $overload(
```

Comment:
Should this be `overloadOf` so that it reads more clearly?  I don't have a strong opinion here.

Reply:
I recall people liking `@overloads(generalOp)` as in "it overloads generalOp", but I'm fine with `@overload` or `@overloadOf` too.

Reply:
I mean, it is not entirely impossible that I had inspiration from [elsewhere](https://docs.python.org/3/library/typing.html#typing.overload) when I picked the name. The main issue I have with the trailing `s` is that it becomes a plural-s.

Reply:
Prior art is good to see, I don't think the slight differences in name will make a huge difference either way so we can keep it as it is if you don't prefer any of the other options.